### ISSUE TITLE
NEW: BSS

### ISSUE THREAD

**Issue by pburdet** [vidartf/hyperspyUI#4]
I'm sure blind source separation is in the pipeline, I just like to mention it.

**Comment by vidartf** [comment]
I don't really have that much experience with BSS, but I made a quick attempt from what I could glance of the behavior from the code (in #8). Does it work as expected? If not, how should it change?

**Comment by pburdet** [comment]
The general work flow for BSS is: 

1. Apply PCA
2. Chose the numebr of component from the PCA result
3. Apply BSS
4. Look at the result (loadings and factors) (never use to reconstruct a new signal: give the same result than PCA)

```python

>>> s.decomposition()
>>>s.blind_source_separation(3)
>>>s.plot_bss_results()

```

**Comment by vidartf** [comment]
@pburdet Latest commits in #8 should fix it (#8 runs with hyperspy/master), yes?

**Comment by pburdet** [comment]
That perfects. Thanks
